### DISCUSSION TITLE
Visualize overall distribution of all genes

### DISCUSSION THREAD

**Discussion by royfrancis** [satijalab/seurat#5081] [Q&A] [answered]
I would like to plot the overall distribution of all genes. For example, `VlnPlot()` for all genes.  I know how to do this manually, but is there a plotting function for this, that I might have missed?

**Comment by timoast** [comment, marked as answer]
I'm not sure what you mean, can you clarify or give an example?

**Reply by royfrancis** [reply]
A violin plot showing expression distribution for all genes.

```r
library(Seurat)
library(ggplot2)
x <- data.frame(value=as.vector(as.matrix(GetAssayData(pbmc_small,slot="data"))))
x$sample <- "sample"
ggplot(x,aes(x=sample,y=value))+
  geom_violin()
```

![vln](https://user-images.githubusercontent.com/5083500/133820002-85da1c17-a821-4711-a6bd-54c0ba8bc498.png)

So, I was expecting something like this with the Seurat violin plot function

```r
Seurat::VlnPlot(pbmc_small)
Error in ExIPlot(object = object, type = ifelse(test = split.plot, yes = "splitViolin",  : 
  argument "features" is missing, with no default
```

without any features specified to display all features.

**Reply by timoast** [reply]
We don't have a function to do that. Since it involves converting the sparse matrix to a dense representation, this could require huge amounts or memory for larger datasets, so I don't think we will add this as a function in Seurat, especially since it can be done easily as you've shown
